File: packages/expo-media-library/ios/next/objects/album/AlbumMetadata.swift
import ExpoModulesCore

struct AlbumMetadata: Record {
  @Field var id: String
  @Field var title: String
  @Field var type: AlbumType?

  init() {}

  init(id: String, title: String, type: AlbumType?) {
    self.id = id
    self.title = title
    self.type = type
  }
}

File: packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/next/records/AlbumMetadata.kt
package expo.modules.medialibrary.next.records

import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record
import expo.modules.kotlin.types.OptimizedRecord

@OptimizedRecord
data class AlbumMetadata(
  @Field val id: String,
  @Field val title: String,
  // iOS only field
  @Field val type: String? = null
) : Record
